Assets/Scripts/Diablerie/Engine/IO/Wav.cs

```diff
@@ -130,6 +130,19 @@ private static float ReadSample(BinaryReader reader)
         return BytesToFloat(byte1, byte2);
     }
 
+    private static void ReadSample(BinaryReader reader, float[] sampleBuffer, int startIndex, int sampleCount)
+    {
+        byte[] buffer = new byte[sampleCount * 2];
+        reader.BaseStream.Read(buffer, 0, buffer.Length);
+        for (int i = 0; i < buffer.Length; i += 2)
+        {
+            byte byte1 = buffer[i];
+            byte byte2 = buffer[i + 1];
+            short s = (short)((byte2 << 8) | byte1);
```

Comment:
Let's reuse `BytesToFloat` here to avoid the transformation logic duplication, please.

Reply:
Maybe we shall test the performances of the following two solutions:
1. reuse `BytesToFloat` method;
2. directly insert body of `BytesToFloat`;

After performance test, choose the fastest one.

Reply:
Do you expect significant function call overhead? In this case, I like your idea to test both of them. If the inlined version works faster I think it's a reasonable cost to duplicate the code.